File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
using namespace std;
 
class Stack {
public:
	int stack[10000]; 
	int size; 
 
	Stack(){ 
        size = 0;
         }
 
	void push(int data){
		stack[size] = data;
		size += 1;
	}
 
	bool empty(){
		if (size == 0)
		{
			return true;
		}
		else
		{
			return false;
		}
	}
 
	int pop() {
		if (empty())
		{
			return -1;
		}
		else
		{
			stack[size - 1] = 0;
			size -= 1;
			return stack[size];
		}
	}
 
	int top() {
		if (empty())
		{
			return -1;
		}
		else
		{
			return stack[size - 1];
		}
	}
	
 
};
 
int main() {

	int N;
	cin >> N;
 
	Stack s;
 
	for (int i = 0; i < N; i++)
	{
		string cmd; 
		cin >> cmd; 
 
		if (cmd == "push")
		{
			int data;
			cin >> data;
			s.push(data);
		}
		else if (cmd == "pop")
		{
			cout << (s.empty() ? -1 : s.top()) << '\n';
			if (!s.empty())
			{
				s.pop();
			}
		}
		else if (cmd =="size")
		{
			cout << s.size << "\n";
		}
		else if (cmd =="empty")
		{
			cout << s.empty() << "\n";
		}
		else if (cmd == "top")
		{
			cout << s.top() << "\n";
		}
	}
 
	return 0;
 
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Stack<Integer> stack = new Stack<>();
        int N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            String[] input = br.readLine().split(" ");
            switch (input[0]) {
                case "push":
                    stack.push(Integer.parseInt(input[1]));
                    break;
                case "pop":
                    bw.write((stack.isEmpty() ? -1 : stack.pop()) + "\n");
                    break;
                case "size":
                    bw.write(stack.size() + "\n");
                    break;
                case "empty":
                    bw.write((stack.isEmpty() ? 1 : 0) + "\n");
                    break;
                case "top":
                    bw.write((stack.isEmpty() ? -1 : stack.peek()) + "\n");
                    break;
            }
        }
        bw.flush();
        bw.close();
    }
}
